contracts/smartdeploy/src/registry/contract.rs

```diff
@@ -32,9 +31,29 @@ pub struct DeployEventData {
     deployer: Address,
     contract_id: Address,
 }
+#[contracttype(export = false)]
+pub struct ContractRegistry(pub Map<String, ContractState>);
 
 #[contracttype(export = false)]
-pub struct ContractRegistry(pub Map<String, Address>);
+#[derive(Clone)]
+pub enum ContractState {
```

Comment:
The name is a bit confusing. It's more of a contract type. So changing the name and the cases names would make it clearer.

Reply:
I changed the enum like that:
```rust
pub enum ContractType {
    ContractById(Address),
    ContractByIdAndOwner(Address, Address),
}
```